core/trino-main/src/main/java/io/trino/server/security/oauth2/TokenPairSerializer.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.server.security.oauth2;
+
+import io.trino.server.security.oauth2.OAuth2Client.Response;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public interface TokenPairSerializer
+{
+    TokenPairSerializer PLAIN_TOKEN_SERIALIZER = new TokenPairSerializer()
+    {
+        @Override
+        public TokenPair deserialize(String token)
+        {
+            return TokenPair.accessToken(token);
+        }
+
+        @Override
+        public String serialize(TokenPair tokenPair)
+        {
+            return tokenPair.getAccessToken();
+        }
+    };
+
+    TokenPair deserialize(String token);
+
+    String serialize(TokenPair tokenPair);
+
+    class TokenPair
+    {
+        private final String accessToken;
+        private final Optional<String> refreshToken;
+
+        private TokenPair(String accessToken, Optional<String> refreshToken)
+        {
+            this.accessToken = requireNonNull(accessToken, "accessToken is nul");
+            this.refreshToken = requireNonNull(refreshToken, "refreshToken is null");
+        }
+
+        public static TokenPair accessToken(String accessToken)
+        {
+            return new TokenPair(accessToken, Optional.empty());
+        }
+
+        public static TokenPair fromOAuth2Response(Response tokens)
+        {
+            requireNonNull(tokens, "tokens is null");
+            return new TokenPair(tokens.getAccessToken(), tokens.getRefreshToken());
+        }
+
+        public static TokenPair accessAndRefreshTokens(String accessToken, String refreshToken)
+        {
+            return new TokenPair(accessToken, Optional.of(requireNonNull(refreshToken, "refreshToken is null")));
```

Comment:
Sure, I would just rather want that to be verified by the caller, so that the caller would use another method like TokenPair.accessToken(). I've added however method to handle Response from OAuth2Client, so this what you are saying should be handled there - I will double check that this is the case.

Reply:
If the calling code is choosing a method based on an optional field then there should be a single method to handle both cases, there's no real need for the calling code to care if the refresh token is present or not.

Reply:
Ok you are right, It is a bug waiting to happen.